content/React Query and management of server state.md

```diff
@@ -0,0 +1,75 @@
+
+## Introduction to React query API
+Although React Query has only two hooks, it is highly configurable in all aspects, since the retry delay time of the queries until what is the interval time of re-fetch in background. 
+But let's start with the most basic API that is well documented in [their repository](https://github.com/tannerlinsley/react-query)
+``` javascript 
+const { status, data, error } =  useQuery('movies', fetchTodoList);
+```
+This hook provided us the status of the fetch(loading, error, or success), data and error if are defined. Up to this point is a pretty normal hook for fetching information, but as we see the query is related to a key(in this case `movies`), this is a unique-global key that is used for associate the query information between the components, in this way we can re-use this hook use the same information in anywhere of the DOM tree. If you follow the classic redux action-reducer cycle for fetching data, **this hooks it will save up a lot of repetitive code**.
+
+Among other things, React Query also allows us to filter the information with a simple system.
+```javascript
+// An individual movie
+useQuery(['movies', 5], ...);
+
+// A list of movies that are "released"
+useQuery(['movies', { type:  'released' }], ...);
+```
+And also, for contemplate the case that the client-side alter the server state React Query introduce the concept of **mutations**(well known for GraphQL developers), let's make a quick example
+
+```javascript
+const [mutate, mutationState] = useMutation(patchMovies, {
+   onSuccess: newMovie => {
+     // Update `movies` while refetching all the list on background
+     queryCache.refetchQueries(movies");
+     queryCache.setQueryData(["movies", { id: newMovie.id }], newMovie);
+   }
+ });
+
+ const onSave = newMovie => mutate(newMovie);
+```
+With `onSave` function we're re-fetching the information on the background while immediately change the specific information(in this case the new movie) in the user interface.
+
+## Manage cached data
+Save data that we fetched and how we saved is an important aspect of our application, improves the sensation of speed in our interfaces, and avoiding hit the backend all the time.
+One of the more important aspects of React Query that sets it apart from other libraries like [SWR](https://github.com/zeit/swr) is the strategy and mechanism to manage the cache.
+For managing cache, there are a lot of alternatives and strategies we can adopt an in more of the cases depends a lot of the problem we need to resolve it, but for general cases, and more particularly for frontend cases, like pagination, infinite scroll, or just show information, [stale-while-revalidate](https://web.dev/stale-while-revalidate/) strategy is an excellent choice.
+This strategy consists, as the name says, revalidate the information(fetching) while the old data is showed to the user. Let's put an example to make it clearer
```

Comment:
`This strategy consists _in_, as the name says, _revalidating_ the information (fetching) while the old data is _shown_ to the user.`